docs/source/starter/style_guide.rst

```diff
@@ -147,26 +153,27 @@ In practice, this code looks like:
 
 Forward vs training_step
 ========================
-We recommend using forward for inference/predictions and keeping training_step independent
+
+We recommend using forward for inference/predictions and keeping ``training_step`` independent.
 
 .. code-block:: python
 
     def forward(self, x):
         embeddings = self.encoder(x)
 
 
-    def training_step(self):
-        x, y = ...
+    def training_step(self, batch, batch_idx):
+        x, y = batch
         z = self.encoder(x)
         pred = self.decoder(z)
         ...
 
-However, when using DataParallel, you will need to call forward manually
+However, when using DataParallel, you will need to call forward manually:
 
 .. code-block:: python
 
-    def training_step(self):
-        x, y = ...
+    def training_step(self, batch, batch_idx):
+        x, y = batch
         z = self(x)  # < ---------- instead of self.encoder(x)
         pred = self.decoder(z)
         ...
```

Comment:
should we recommend users to implement forward then to avoid a special carveout for DataParallel? it's not intuitive why DataParallel requires this

Reply:
DP is a special case worth highlighting. Advising implementing a forward method always might not be good for this section.